hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/BlockCacheKey.java

```diff
@@ -39,24 +39,16 @@ public class BlockCacheKey implements HeapSize, java.io.Serializable {
    * @param hfileName The name of the HFile this block belongs to.
    * @param offset    Offset of the block into the file
    */
-  public BlockCacheKey(String hfileName, long offset) {
-    this(hfileName, offset, true, BlockType.DATA);
-  }
-
-  public BlockCacheKey(String hfileName, long offset, boolean isPrimaryReplica,
-    BlockType blockType) {
-    this.isPrimaryReplicaBlock = isPrimaryReplica;
-    this.hfileName = hfileName;
-    this.offset = offset;
-    this.blockType = blockType;
+  public BlockCacheKey(Path hfilePath, long offset) {
+    this(hfilePath, offset, true, BlockType.DATA);
   }
```

Comment:
Can we just leave these and avoid having to touch every single existing test? As having a complete path is only relevant for the TIME_RANGE data tiering, I think it's fine to allow non complete paths when TIME_RANGE data tiering is not in use.